doc/api-design.yml

```diff
@@ -0,0 +1,903 @@
+                      type: string
+                    wrong_answer2:
+                      type: string
+                    wrong_answer3:
+                      type: string
+        '400':
+          description: Invalid input
+  /quizzes:
+    post:
+      summary: Create a new quiz
+      tags:
+        - Quizzes
+      security:
+        - bearerAuth: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                title:
+                  type: string
+                description:
+                  type: string
+                image:
+                  type: string
+                  format: base64
+                questions:
+                  type: array
+                  items:
+                    type: object
+                    properties:
+                      question_type:
+                        type: string
+                        enum: [english_to_turkish, turkish_to_english, english_to_sense]
+                      word:
+                        type: string
+                      correct_answer:
+                        type: string
+                      wrong_answers:
+                        type: array
+                        items:
+                          type: string
+              required:
+                - title
+                - questions
+      responses:
+        '201':
+          description: Quiz created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Quiz'
+        '400':
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    get:
+      summary: Get list of quizzes
+      tags:
+        - Quizzes
+      security:
+        - bearerAuth: []
+      parameters:
+        - name: page
+          in: query
+          schema:
+            type: integer
+          description: Page number
+        - name: limit
+          in: query
+          schema:
+            type: integer
+          description: Number of items per page
+        - name: filter
+          in: query
+          schema:
+            type: string
+            enum: [own, others, all]
+          description: Filter quizzes by ownership (own, others, or all)
+        - name: difficulty
+          in: query
+          schema:
+            type: string
+            enum: [a1,a2,b1,b2,c1,c2]
+          description: Filter quizzes by difficulty
+      responses:
+        '200':
+          description: A list of quizzes
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  quizzes:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Quiz'
```

Comment:
We also need the description on the listing page, but it currently doesn't exist on `GET /quizzes` endpoints response

Reply:
We also need image on listing response